### ISSUE TITLE
simd detects neon support incorrectly for ARMv5

### ISSUE THREAD

**Issue by michaelolbrich** [mesonbuild/meson#5973]
Just checking if the compiler accepts `-mfpu=neon` does not work in all cases.
An ARMv5 soft-float compiler accepts the flag without error for a basically empty source file.
However, compiling actual code fails.
For example building pulseaudio fails with:
```
...include/arm_neon.h:31:2: error: #error "NEON intrinsics not available with the soft-float ABI. Please use -mfloat-abi=softfp or -mfloat-abi=hard"
```
Maybe trying to build `arm_neon.h` would make sense as a check.

**Comment by jpakkane** [comment]
Is it guaranteed to be available on all compiler toolchains and operating systems that support ARM?

**Comment by michaelolbrich** [comment]
From a hardware perspective, NEON is not available at all on ARMv5 and ARMv6 (e.g. Raspberry Pi) and it's optional for ARMv7.
From a compiler ABI perspective, NEON requires hardware floating point in general. I think there are 3 possible compiler configurations:
- `-mfloat-abi=soft` no hardware floating point at all
- `-mfloat-abi=softfp` same ABI as the option as above but with  hardware floating point
- `-mfloat-abi=hard` hardware floating point but with a different ABI

I think the last two should be able to compile the code without error (seems to work for pulseaudio).

For the first option, you might be able to compile the code by adding `-mfloat-abi=softfp`. Note, that you cannot add this unconditionally because you cannot mix code compiled with `-mfloat-abi=softfp` and `-mfloat-abi=hard`.
Also, `-mfloat-abi=soft` is only used for ARMv5 these days so building in the NEON code that is never used anyways is a waste.

As a side note, as a embedded developer, I'd like an option to overwrite the list of supported simd types. The compiler usually supports more than the hardware, and this would be great to avoid unnecessary code. This would also be useful in case the auto detection got it wrong and the code fails to compile.

**Comment by dcbaker** [comment]
Closed in favor of #6361
